tarodb/asset_minting.go

```diff
@@ -255,18 +255,34 @@ func (a *AssetMintingStore) CommitMintingBatch(ctx context.Context,
 				"batch: %w", err)
 		}
 
-		// Now that our minting batch is in place, which defences the
+		// Now that our minting batch is in place, which references the
 		// internal key inserted above, we can create the set of new
 		// seedlings.
 		for _, seedling := range newBatch.Seedlings {
-			err := q.InsertAssetSeedling(ctx, AssetSeedlingShell{
+			dbSeedling := AssetSeedlingShell{
 				BatchID:         batchID,
 				AssetName:       seedling.AssetName,
 				AssetType:       int16(seedling.AssetType),
 				AssetSupply:     int64(seedling.Amount),
 				AssetMeta:       seedling.Metadata,
 				EmissionEnabled: seedling.EnableEmission,
-			})
+			}
+
+			// If this seedling is being issued to an existing
+			// group, we need to reference the genesis that
+			// was first used to create the group.
+			if seedling.HasGroupKey() {
+				genesisID, err := fetchGenesisID(
+					ctx, q, *seedling.GroupInfo.Genesis,
+				)
+				if err != nil {
+					return err
+				}
+
+				dbSeedling.GroupGenesisID = sqlInt32(genesisID)
```

Comment:
Alternatively, couldn't we point to the group key id itself?

Reply:
IIRC with the genesis we can easily find the correct group but not the reverse. We want the group genesis linked to the seedling so we can later use it (+ raw key info) for signing when converting the seedling to a sprout.